spec/unit/shoulda/matchers/active_model/validate_numericality_of_matcher_spec.rb

```diff
@@ -1141,6 +1216,21 @@ def define_model_validating_numericality(options = {})
     end
   end
 
+  def define_model_validating_numericality_of_virtual_attribute(options = {})
+    virtual_attr = options.delete(:virtual_attribute) do
+      virtual_attribute_name
+    end
+
+    define_model 'Example' do |model|
```

Comment:
An ActiveModel model would not have tripped this problem with virtual attributes though because it would not have responded to columns_hash in the first place. That check was already there. This specifically happens with an ActiveRecord model using virtual attributes as in has_secure_password.